nifi-commons/nifi-processor-utilities/src/main/java/org/apache/nifi/processor/util/bin/Bin.java

```diff
@@ -121,8 +125,9 @@ public boolean offer(final FlowFile flowFile, final ProcessSession session) {
             final String countValue = flowFile.getAttribute(fileCountAttribute);
             final Integer count = toInteger(countValue);
             if (count != null) {
-                this.maximumEntries = Math.min(count, this.maximumEntries);
-                this.minimumEntries = this.maximumEntries;
+                int currentMaxEntries = this.maximumEntries;
+                this.maximumEntries = Math.min(count, currentMaxEntries);
+                this.minimumEntries = currentMaxEntries;
```

Comment:
Not sure why it was necessary (other then style), but overall PR looks good.
+1